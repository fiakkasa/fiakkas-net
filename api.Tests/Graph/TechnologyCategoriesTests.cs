using HotChocolate.Execution;

namespace api.Tests.Graph;

public class TechnologyCategoriesTests(GraphFixture fixture) : IClassFixture<GraphFixture>
{
    [Fact]
    public async Task TechnologyCategories_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
            """
            {
              technologyCategories {
                totalCount
                nodes {
                  __typename
                  createdAt
                  href
                  id
                  title
                  updatedAt
                  version
                  ... on InformationTechnologyCategory {
                    createdAt
                    href
                    id
                    internalId
                    title
                    updatedAt
                    version
                    portfolioCategories {
                      totalCount
                    }
                    portfolioCustomers {
                      totalCount
                    }
                    portfolioItems {
                      totalCount
                    }
                  }
                  ... on SoftwareDevelopmentCategory {
                    createdAt
                    href
                    id
                    internalId
                    title
                    updatedAt
                    version
                    portfolioCategories {
                      totalCount
                    }
                    portfolioCustomers {
                      totalCount
                    }
                    portfolioItems {
                      totalCount
                    }
                  }
                  ... on UnknownTechnologyCategory {
                    createdAt
                    href
                    id
                    internalId
                    title
                    updatedAt
                    version
                    portfolioCategories {
                      totalCount
                    }
                    portfolioCustomers {
                      totalCount
                    }
                    portfolioItems {
                      totalCount
                    }
                  }
                }
              }
            }
            """);

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }
}
