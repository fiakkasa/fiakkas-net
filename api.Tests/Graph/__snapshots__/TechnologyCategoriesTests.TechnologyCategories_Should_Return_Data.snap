{
  "data": {
    "technologyCategories": {
      "totalCount": 2,
      "nodes": [
        {
          "__typename": "SoftwareDevelopmentCategory",
          "createdAt": "2024-01-01T00:00:00.000Z",
          "href": "/test",
          "id": "U29mdHdhcmVEZXZlbG9wbWVudENhdGVnb3J5Ovkrg8rLtzFMv40A+Honb+M=",
          "title": "Title",
          "updatedAt": null,
          "version": 1,
          "internalId": "ca832bf9-b7cb-4c31-bf8d-00f87a276fe3",
          "portfolioCategories": {
            "totalCount": 1
          },
          "portfolioCustomers": {
            "totalCount": 1
          },
          "portfolioItems": {
            "totalCount": 1
          }
        },
        {
          "__typename": "InformationTechnologyCategory",
          "createdAt": "2024-01-01T00:00:00.000Z",
          "href": "/test",
          "id": "SW5mb3JtYXRpb25UZWNobm9sb2d5Q2F0ZWdvcnk6mj7EbCsxI0m4kOlmuBaO7g==",
          "title": "Title",
          "updatedAt": null,
          "version": 1,
          "internalId": "6cc43e9a-312b-4923-b890-e966b8168eee",
          "portfolioCategories": {
            "totalCount": 0
          },
          "portfolioCustomers": {
            "totalCount": 0
          },
          "portfolioItems": {
            "totalCount": 0
          }
        }
      ]
    }
  }
}
